src/pages/PlayTimeAttack/hooks/index.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from "react";
+import { useTimeAttackStore } from "../../../store/useTimeAttackStore";
+import { useUserStore } from "../../../store/useUserStore";
+import { useSocketRefStore } from "../../../store/useSocketRefStore";
+import {
+	Event,
+	Mode,
+	type Player,
+} from "../../../proto/game/resources/game_pb";
+// import { useNavigate } from "react-router-dom";
+
+// const gameDuration = 11;
+
+export const usePlayTimeAttack = () => {
+	//   const navigate = useNavigate();
+	const setClearTime = useTimeAttackStore((state) => state.setClearTime);
+	const { name: userName } = useUserStore();
+	const { eventSend } = useSocketRefStore();
```

Comment:
```suggestion
	const  eventSend  = useSocketRefStore((state)=>state.eventSend);
```
https://zustand.docs.pmnd.rs/guides/auto-generating-selectors
セレクタを使って書くことが公式からも推奨されているので、その書き方でお願いします